packages/babel-parser/src/parser/comments.js

```diff
@@ -26,22 +26,31 @@ export type CommentWhitespace = {
   trailingNode: Node | null,
   containingNode: Node | null,
 };
+
 /**
- * Merge comments with node's trailingComments or assign comments to be
- * trailingComments. New comments will be placed before old comments
+ * Merge comments with node's ${type}Comments or assign comments to be
+ * ${type}Comments. New comments will be placed before old comments
  * because the commentStack is enumerated reversely.
  *
+ * @param {"leading" | "inner" | "trailing"} position
  * @param {Node} node
  * @param {Array<Comment>} comments
  */
-function setTrailingComments(node: Node, comments: Array<Comment>) {
-  if (node.trailingComments === undefined) {
-    node.trailingComments = comments;
+function setComments(
+  position: "leading" | "inner" | "trailing",
+  node: Node,
+  comments: Array<Comment>,
+) {
+  if (node[`${position}Comments`] === undefined) {
+    node[`${position}Comments`] = comments;
   } else {
-    node.trailingComments.unshift(...comments);
+    node[`${position}Comments`].unshift(...comments);
   }
 }
 
+const setLeadingComments = setComments.bind(null, "leading");
```

Comment:
`Function#bind` is slow, I would not oppose to implement `setLeadingComments`, too. They are now complete.